modules/dnn/src/layers/scale_layer.cpp

```diff
@@ -29,7 +29,7 @@ class ScaleLayerImpl CV_FINAL : public ScaleLayer
     {
         setParamsFrom(params);
         hasBias = params.get<bool>("bias_term", false);
-        axis = params.get<int>("axis", 1);
+        axis = (params.get<int>("axis", 1) > 0) ? params.get<int>("axis", 1) : 1;
```

Comment:
Is this correct? Why can't axis be negative?

Reply:
Good point, Thanks.

Reply:
@YashasSamaga, please have a look now.

Reply:
I think you're looking for this:

https://github.com/opencv/opencv/blob/ca23c0e630ba9c4fc147c4913d6238f5de3bbcd9/modules/dnn/src/layers/fully_connected_layer.cpp#L118

Reply:
@YashasSamaga not exactly, but yeah

```
axis = clamp(axis , inpBlob.dims) ? clamp(axis , inpBlob.dims) : 1;
CV_Assert(axis > 0 && axis < inpBlob.dims);
```
this would work. 

actually here in this loop, `numWeightsPerSample` is total weights excluding batchsize, so we need to check with axis >= 1.
```
for (endAxis = axis + 1; endAxis <= inpBlob.dims; ++endAxis)
        {
            if (total(inpShape, axis, endAxis) == numWeightsPerSample)
                break;
        }
endAxis = (numWeightsPerSample == 1) ? 1 : endAxis;
```